lib/app/admin/data-reset.js

```diff
@@ -0,0 +1,212 @@
+var q = require('q');
+var async = require('async');
+var _ = require('lodash');
+// default list of types of objects to remove, can be overriden when passed in request body
+// order of the list is important for building relations between them, parent items should be listed first
+var DEFAULT_ITEM_LIST = [
+  'messages',
+  'result',
+  'workflows',
+  'workorders',
+  'group',
+  'membership',
+  'file'
+];
+var DEFAULT_LIST_RELATIONS = {
+  workorders: 'workflows',
+  membership: 'group'
+};
+
+/**
+ * Rebuilds data in the solution - first it removes all items synchronously using mediator ':delete' topic,
+ * this is followed by ':create' topic using seed data located in ./data-demo
+ * @param {Object} mediator - mediator object used for communication
+ * @param {Array} removeList - list of element names e.g. DEFAULT_ITEM_LIST
+ * @returns {Promise} Promise, resolved if all items are successfully rebuilt.
+ */
+module.exports = function rebuildData(mediator, removeList) {
+  var deferred = q.defer();
+  var list = (removeList) ? removeList : DEFAULT_ITEM_LIST;
+  deleteData(mediator, list)
+    .then(function() {
+      buildData(mediator, list);
```

Comment:
If `buildData` returns a promise, then you don't need the `deferred`

You can just have 

```
return deleteData(mediator, list).then(buildData(mediator, list));
```

This will return a promise to the function that called it. The function will look much cleaner that way.